### PR TITLE
feat: add view for translation input

### PR DESCRIPTION
**Pull request by siddarth2824**
## Changes
- Add views to allow users to add translation input
- Link to [view](https://www.figma.com/design/wNcuzzF3ciL8f6xYNvGZgK/Form-Exploration?node-id=11536-21618&node-type=frame&t=0av2bQA7SjvvIw2k-0)

## Before & After Screenshots
| Description | Before | After | 
| ----------- | ------- | ----- |
|Translation container for users to input translations for base form fields| |Screenshot 2024-11-18 at 10 21 18 AM|
|Translation container for users to input their translations for form fields with options| |Screenshot 2024-11-18 at 10 06 32 AM|
|Translation container for users to input translations for table container| |Screenshot 2024-11-18 at 10 19 48 AM|

### CODE REVIEW

File: frontend/src/features/admin-form/settings/components/MultiLanguageSection/EndPageTranslationContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+export const EndPageTranslationsContainer = ({
+  endPage,
+  capitalisedLanguage,
+  unicodeLocale,
+}: EndPageTranslationsContainerProps) => {
+  if (!endPage) return null
+
+  const hasParagraph = endPage.paragraph && endPage.paragraph.trim() !== ''
```
Comment (KenLSM) [inline frontend/src/features/admin-form/settings/components/MultiLanguageSection/EndPageTranslationContainer.tsx]: /nit can be terser
```suggestion
  const hasParagraph = && endPage.paragraph?.trim() !== ''
```

File: frontend/src/features/admin-form/settings/components/MultiLanguageSection/mutations/useTranslationLogic.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+      }
+    }
+
+    if (isStartPageTranslations && formStartPage) {
+      const updatedFormStartPage = updateFormStartPageTranslations({
+        data: formStartPage,
+        language,
+        updatedParagraphTranslation,
+      })
+      console.log(updatedFormStartPage)
```
Comment (KenLSM) [inline frontend/src/features/admin-form/settings/components/MultiLanguageSection/mutations/useTranslationLogic.ts]: A couple of stray `console.logs` here

File: frontend/src/features/admin-form/settings/SettingsMultiLangPage.tsx [hunk 3]
```diff
@@ -1,15 +1,40 @@
...
 export const SettingsMultiLangPage = (): JSX.Element => {
   const [searchParams] = useSearchParams()
-  const unicodeLocale = searchParams.get('unicodeLocale')
+  const unicodeLocale = searchParams.get(UNICODE_LOCALE)
+  const translationInput = searchParams.get(TRANSLATION_INPUT)
+  const isTranslationInput = !_.isNull(translationInput)
+  const isEndPageTranslationInput = translationInput === 'endPage'
+  const isStartPageTransltionInput = translationInput === 'startPage'
 
   return (
```
Comment (KenLSM) [inline frontend/src/features/admin-form/settings/SettingsMultiLangPage.tsx:26]: Here's my attempt at separating out the the nested ternary operators out from the render function -- I generally avoid deeply nested render functions as they can be hard to read.

```suggestion
  // Request user to select a language
  if (!unicodeLocale) {
    return <MultiLanguageSection />
  }

  // Request user to select fields to translate
  if (!isTranslationInput) {
    return <TranslationListSection language={unicodeLocale} />
  }

  return (
    <TranslationSection
      language={unicodeLocale}
      formFieldNumToBeTranslated={
        isEndPageTranslationInput || isStartPageTransltionInput
          ? -1
          : _.toNumber(translationInput)
      }
      isEndPageTranslations={isEndPageTranslationInput}
      isStartPageTranslations={isStartPageTransltionInput}
    />
  )
```

Comment (siddarth2824) [reply frontend/src/features/admin-form/settings/SettingsMultiLangPage.tsx:26]: Thanks for this suggestion. It's a really neat way of handling multiple logics.

**Review by KenLSM — APPROVED** [review]
LGTM

**Pull request merged**